src/libs/Navigation/Navigation.js

```diff
@@ -186,6 +191,23 @@ function isActiveRoute(routePath) {
     return getActiveRoute().substring(1) === routePath;
 }
 
+/**
+ * @returns {Promise}
+ */
+function isNavigationReady() {
```

Comment:
Just a minor comment that doesn't have to be a blocker .... we only have this code because of a [known bug](https://github.com/react-navigation/react-navigation/issues/9170#issuecomment-898831026) in react-navigation. If possible, we should create a minimal reproduction and a fresh issue in the react-navigation repo, and see if we can either submit a PR to fix it or hire the maintainer to fix it.

Reply:
I'm not sure it would help us though as the navigation becomes available and unavailable when conditionally rendering the navigators e.g. in Neil's case we are opening the app while logged in then logging out then logging in and trying to navigate. Unless the `onReady()` callback can be called multiple times while changing the root navigator. Not really sure what the expected behavior is there, but seems like the maintainer implied it was [**not a bug**](https://github.com/react-navigation/react-navigation/issues/9170#issuecomment-750289596) and suggested we stop conditionally rendering the navigators (easier said than done of course 😄 ).

Reply:
Could we conditionally render the Public and Auth screens rather than the navigators? They both use a stack navigator.

Reply:
> Could we conditionally render the Public and Auth screens rather than the navigators

I looked into this a while back, while working on this exact same flow. It was ... problematic. I don't remember everything about it, but here's [some context](https://github.com/react-navigation/react-navigation/issues/9987). Not sayin' it's impossible, only that I gave up. 😅

Reply:
Yeah I think it will be a big change, but if it solves all of our "navigator is not ready" issues then it could be worth looking into... eventually!